packages/ipns/src/utils/dns.ts

```diff
@@ -0,0 +1,99 @@
+import { CodeError } from '@libp2p/interfaces/errors'
+import * as isIPFS from 'is-ipfs'
+import type { ResolveDnsLinkOptions } from '../index.js'
+
+export interface Question {
+  name: string
+  type: number
+}
+
+export interface Answer {
+  name: string
+  type: number
+  TTL: number
+  data: string
+}
+
```

Comment:
Is this supposed to be a conjoined object representing a response from either rfc1035 or rfc8427 ?

```suggestion
/**
 * This type represents a DNS query response for either rfc1035 or rfc8427 
 * @see https://www.ietf.org/rfc/rfc1035.txt
 * @see https://www.ietf.org/rfc/rfc8427.txt
 */
```

If not, we should specify what's different.

Reply:
It looks like this is only used inside of `dns-over-https`, so maybe we can clarify that this is used to convert an rfc1035 response into a format similar to `dns-json-over-https` (or whatever it's actually doing)

Reply:
ignoring this for now.